src/components/activity/ActivityRegistrator.tsx

```diff
@@ -0,0 +1,194 @@
+
+
+  const [values, modifyValues] = Storage.useValues();
+  const [activities, modifyActivities] = Storage.useActivities();
+  
+  const [fromTime, setFromTime] = React.useState(getCurrentTimeRounded(0, steps));
+  const [toTime, setToTime] = React.useState(getCurrentTimeRounded(1, steps));
+  
+  const [date, setDate] = React.useState(new Date());
+  
+  const [choise, setChoise] = React.useState(defaultChoise);
+  const [activityText, setActivityText] = React.useState('');
+
+  const [importance, setImportance] = React.useState(5); 
+  const [enjoyment, setEnjoyment] = React.useState(5);
+
+
+  // Find topics to choose from depending on input icon
+  const addTopicEntries = (topics: any) => {
+    for(let topicIndex = 0; topicIndex < topics.length; ++topicIndex) {
+      for(let entryIndex = 0; entryIndex < topics[topicIndex].entries.length; ++entryIndex){
+        const entry = topics[topicIndex].entries[entryIndex];
+
+        // If icons match, include it in choises
+        if (entry.icon == route.params.icon) {
+          choises.push({
+            value: entry.text,
+            isDefault: false
+          });
+        }
+      }
+    }
+  };
+
+  // Go through all categories in values
+  addTopicEntries(values.responsibilities);
+  addTopicEntries(values.relations);
+  addTopicEntries(values.enjoyment);
+  addTopicEntries(values.health);
+  addTopicEntries(values.work);
+
+
+  const onConfirm = () => {
+    // Check if custom text has been entered
+    const entryText = choise.isDefault ? activityText : choise.value;
+    
+    // Create entry from information entered by user
+    const entry = {
+      text: entryText,
+      icon: route.params.icon,
+      person: '', // TODO: link to value based on choise
+      meaningful: importance,
+      entertaining: enjoyment,
+    };
+
+    // If toTime returns 0 hours its the next day, count it as 24.
+    let toHour = toTime.getHours();
+    if (toHour === 0) {
+      toHour = 24;
+    }
+
+    // Splitting at 'T' to extract only the date from the ISO string.
+    const isoDateString = date.toISOString().split('T')[0];
+
+    // Add entry att every applicable hour
+    for(let i = fromTime.getHours(); i < toHour; ++i) {
+      modifyActivities.add(isoDateString, i, entry);
+    }
+
+    // Go back
+    navigation.navigate('Activities', {activityRegistered: true})
+  };
+
+
+  const onCancel = () => {
+    // Go back
+    navigation.navigate('Activities', {activityRegistered: false})
+  };
+
+  // Attempts to dissmiss the keyboard when the 
+  const dismissKeyboard = () => {
+    Keyboard.dismiss();
+  };
+
+  /**
+   * This hook/function saves the auto-scaled height in order to keep that height
+   *  when then keyboard is presented.
+   */
+  const [absHeight, setAbsHeight] = React.useState(-1);
+  const onLayoutSet = (event: any) => {
+    // If absHeight has not been set
+    if(absHeight === -1) {
+      var {x, y, width, height} = event.nativeEvent.layout;
```

Comment:
```suggestion
      const {x, y, width, height} = event.nativeEvent.layout;
```
I assume there is no reason to have `var`

Reply:
Correct, var is not needed. This is a typo from a code example.
However, let has to be used here because of the assignment.